packages/runtime/container-runtime/src/summaryManager.ts

```diff
@@ -317,7 +318,32 @@ export class SummaryManager extends TypedEventEmitter<ISummaryManagerEvents> imp
         }
 
         if (delayMs > 0) {
-            await delay(delayMs);
+            // eslint-disable-next-line prefer-const
```

Comment:
Not sure if it's super clear from current code, but original intent (that I tried to preserve) is to always wait a delay that is coming from startThrottler, and on top of that - delay more if we do not have enough ops and this.initialDelayMs is bigger than delay from throttler. You are short-circuiting throttler delay below.

That said, I'm not sure throttler is needed / helpful / what kind of purpose it has here. Maybe worth looking into history to understand it better? It feels like it's not needed (or if it's needed, I'd love to see comments in the code clearly describing intent)

Reply:
I see. Would you prefer breaking it into 2 delays, a throttler delay that always happens in full, followed by a secondary delay that can be broken out of? I'll try and find out more about the throttler.

Reply:
They are not stacked one on top of the other (i.e. not delay + delay). It's a delay(max(A, B)) where one delay is throttler delay, and another (before your change) is either  0 or initialDelay.
It's less about my preferences :), it's more about what this code intention is.
My preference would be to get rid of throttler delay as I do not understand what problem it solves, but I lack history / context to make this call. I do not remember what telemetry we output before my changes (now we output both numbers to be able to make various analyses), but if we had good telemetry in the past - one way to understand usefulness of throttler delay is to look at telemetry if it every gets involved, and if yes - does this help with the problem we face in those sessions or not.

Reply:
Here's what I see: There used to be an initial delay promise belonging to the SummaryManager. It wrapped a PromiseTimer that got started in the SummaryManager ctor. So it seems like that delay made sure that we didn't try to summarize immediately on SummaryManager creation. Then there's the throttler delay that fluctuates over time as we attempt to spawn summarizers, where the goal seems to be to avoid a death spiral of spawn attempts. @wes-carlson, does that all sound right? It looks like you added the existing throttler mechanism with https://github.com/microsoft/FluidFramework/pull/1562.

Reply:
> where the goal seems to be to avoid a death spiral of spawn attempts

That's correct. From the code:
> // throttle creation of new summarizer containers to prevent spamming the server with websocket connections

Maybe it's not necessary anymore if we are respecting retryAfter, but this is why it was added.

Reply:
Thanks. Which usage of retryAfter are you referring to? And is it true that the "initial delay" was originally supposed to apply just after the creation of the SummaryManager, so the throttling delay was the only delay that applied over the lifetime of the SummaryManager?

Reply:
My own sense, though, is that both the initial delay and the throttling delay should be broken out of if we've accumulated enough ops, so we could reasonably take this change as it is and possibly rethink the delay logic independently.

Reply:
>Which usage of retryAfter are you referring to?

Nothing specific. I think when this was added we wouldn't handle it at all, so maybe it's not necessary anymore.

> And is it true that the "initial delay" was originally supposed to apply just after the creation of the SummaryManager, so the throttling delay was the only delay that applied over the lifetime of the SummaryManager?

I think so. My understanding is that initial delay was just to avoid summarizer load from affecting the performance of the parent container load. The throttler is there to  stop it from spawning new containers in a tight loop if it's immediately hitting an error.